### DISCUSSION TITLE
Using cluster local chartmuseum

### DISCUSSION THREAD

**Discussion by PeteMac88** [fluxcd/flux2#1189] [Q&A]
Hey,
I have a question regarding the proposed way of using s3 helm registries with flux v2. We managed to setup chartmuseum in our cluster as a source for our hem charts and on first glance eveythink worked fine. Then we realized that newer chart version which we push to s3 through our github actions are not present in the cluster until you restart the chartmuseum pod. It seems that chartmuseum is not updating its internal index.yaml when a chart is pushed directly to s3.

Did you come across this problem already since you name using a cluster local chartmuseum as the preferred way to go? We are now using the `Bucket` resource as a source.

**Comment by stefanprodan** [comment]
Chartmuseum has an API for uploading charts, if you use it then it will update the index, see https://github.com/helm/chartmuseum#chart-manipulation

**Reply by PeteMac88** [reply]
Hey thanks for the answer :) . I already knew this but for our use case this is not really an option. We only use chartmuseum to easily fetch the charts from s3 since directly using the Bucket resource is not recommended. We still want to push in our CD processes to our central s3 helm registry and chartmuseum should recognize this.
